### ISSUE TITLE
Migrate parser tests to use ParserAssert

### ISSUE THREAD

**Issue by martint** [trinodb/trino#7520]
Tests in https://github.com/trinodb/trino/blob/master/core/trino-parser/src/test/java/io/trino/sql/parser/TestSqlParser.java that currently rely on `assertExpression` and `assertStatement` need to be updated to use ParserAssert.

Examples:
https://github.com/trinodb/trino/blob/master/core/trino-parser/src/test/java/io/trino/sql/parser/TestSqlParser.java#L234
https://github.com/trinodb/trino/blob/master/core/trino-parser/src/test/java/io/trino/sql/parser/TestSqlParser.java#L1247

**Comment by shandeep00** [comment]
Hello, can I take up this if no one is working on it

**Comment by martint** [comment]
Thanks @shandeep00, go for it!

**Comment by shandeep00** [comment]
@Justin-Kwan I have updated all assertExpression tests, you can work on updating assertStatement tests.
@martint is it okay if we raise 2 separate PR's for this?

**Comment by martint** [comment]
> is it okay if we raise 2 separate PR's for this?

Yes, absolutely. No need to do it all in one shot -- it can be done incrementally for different parts.

**Comment by Justin-Kwan** [comment]
Hmm, I've just about finished updating all of the assertStatement and assertExpression already @shandeep00

**Comment by shandeep00** [comment]
ohh cool, np go ahead and raise a PR. Also, it is better to check if someone is already working on the issue before we start, this will avoid duplicate work.

**Comment by Justin-Kwan** [comment]
thanks, and good idea, I'll make sure to do that next time, my bad. @shandeep00

**Comment by bbreak** [comment]
As this issue has been idle for a few years, I've taken the liberty of filing PR #30280.
